pkg/engine/variables/vars_test.go

```diff
@@ -131,6 +132,50 @@ func Test_subVars_failed(t *testing.T) {
 	}
 }
 
+func Test_ReplacingPathWhenDeleting(t *testing.T) {
+	patternRaw := []byte(`"{{request.object.metadata.annotations.target}}"`)
```

Comment:
Can we add another test to query something like
`{{request.object.metadata.annotations.{{request.object.metadata.annotations.targetnew}}}}`? 
We can have object like this:
```json
"request": {
			"operation": "DELETE",
			"object": {
				"metadata": {
					"name": "curr",
					"namespace": "ns",
					"annotations": {
					  "target": "foo",
					  "targetnew": "target"
					}
				}
			}
```

Reply:
Will do

Reply:
@realshuting done